src/transformers/models/bark/modeling_bark.py

```diff
@@ -0,0 +1,1429 @@
+        )
+
+        attn_output = attn_outputs[0]  # output_attn: output, present_kv, (attn_weights)
+        outputs = attn_outputs[1:]
+
+        intermediary_hidden_states = hidden_states + attn_output
+        intermediary_hidden_states = intermediary_hidden_states + self.mlp(self.ln_2(intermediary_hidden_states))
+
+        if use_cache:
+            outputs = (intermediary_hidden_states,) + outputs
+        else:
+            outputs = (intermediary_hidden_states,) + outputs[1:]
+
+        return outputs  # hidden_states, ((present), attentions)
+
+
+class BarkPreTrainedModel(PreTrainedModel):
+    """
+    An abstract class to handle weights initialization and a simple interface for downloading and loading pretrained
+    models.
+    """
+
+    config_class = BarkConfig
+    supports_gradient_checkpointing = False
+
+    def _init_weights(self, module):
+        if isinstance(module, BarkCausalModule) or isinstance(module, BarkFineAcousticsModule):
+            module.apply(module._init_weights)
+
+    def __init__(self, *inputs, **kwargs):
+        super().__init__(*inputs, **kwargs)
+
+
+class BarkModulePreTrainedModel(PreTrainedModel):
+    """
+    An abstract class to handle weights initialization and a simple interface for downloading and loading pretrained
+    models.
+    """
+
+    config_class = BarkModuleConfig
+    # supports_gradient_checkpointing = True
+
+    def _init_weights(self, module):
+        """Initialize the weights."""
+        if isinstance(module, (nn.Linear,)):
+            # Slightly different from the TF version which uses truncated_normal for initialization
+            # cf https://github.com/pytorch/pytorch/pull/5617
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, nn.Embedding):
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)
+            if module.padding_idx is not None:
+                module.weight.data[module.padding_idx].zero_()
+        elif isinstance(module, nn.LayerNorm):
+            module.bias.data.zero_()
+            module.weight.data.fill_(1.0)
+
+    def _set_gradient_checkpointing(self, module, value=False):
+        if isinstance(module, BarkCausalModule) or isinstance(module, BarkFineAcousticsModule):
+            module.gradient_checkpointing = value
+
+
+# GPT2-like autoregressive model
+class BarkCausalModule(BarkModulePreTrainedModel):
+    # TODO: add code sample when checkpoint is added
+    #
+    # @add_code_sample_docstrings(
+    #    checkpoint=_CHECKPOINT_FOR_DOC,
+    #    output_type=BaseModelOutputWithPast,
+    #    config_class=_CONFIG_FOR_DOC,
+    # )
+
+    def __init__(self, config):
+        super().__init__(config)
+        self.config = config
+
+        # initialize as an autoregressive GPT-like model
+        self.transformer = nn.ModuleDict(
```

Comment:
Unfortunately we can't use `ModuleDict`'s in our `__init__`'s; the consensus is that they're more difficult to understand and track than explicitly defined modules. So what we need to do is switch to explicitly defining each module in the following convention: 

```python
self.wte = nn.Embedding(config.input_vocab_size, config.hidden_size)
self.wpe = nn.Embedding(config.block_size, config.hidden_size)

self.drop = nn.Dropout(config.dropout)

self.layers = nn.ModuleList([BarkBlock(config, is_causal=True) for _ in range(config.num_layers)])

self.ln_f = LayerNorm(config.hidden_size, bias=config.bias)
```

-> this makes the code a lot easier to track (since you can directly refer back to a module `self.wte` rather than an attribute of a module dict (`self.transformer.wte` just corresponds to an entry in a module dict)), but you'll need to update the conversion script to reflect this